palette/moreland/example_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package moreland
+
+import (
+	"log"
+	"math"
+	"os"
+	"testing"
+
+	"github.com/gonum/matrix/mat64"
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/internal/cmpimg"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/plotter"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+	"github.com/gonum/plot/vg/vgimg"
+)
+
+type offsetUnitGrid struct {
+	XOffset, YOffset float64
+
+	Data *mat64.Dense
+}
+
+func (g offsetUnitGrid) Dims() (c, r int)   { r, c = g.Data.Dims(); return c, r }
+func (g offsetUnitGrid) Z(c, r int) float64 { return g.Data.At(r, c) }
+func (g offsetUnitGrid) X(c int) float64 {
+	_, n := g.Data.Dims()
+	if c < 0 || c >= n {
```

Comment:
The convention we use for range checks (where performance is not critical) is `c  < 0 || n <= c` showing that `c` is out side the range (the within is written `0 <= c && c < n`). Also below.